lib/ts/normalisedURLDomain.ts

```diff
@@ -30,48 +30,33 @@ export default class NormalisedURLDomain {
 function normaliseURLDomainOrThrowError(input: string, ignoreProtocol = false): string {
     input = input.trim().toLowerCase();
 
-    try {
-        if (!input.startsWith("http://") && !input.startsWith("https://") && !input.startsWith("supertokens://")) {
-            throw new Error("converting to proper URL");
+    // if the input starts with a . (eg: .domain.tld)
+    if (input.indexOf(".") === 0) {
+        input = input.substring(1);
+    }
+
+    // if the input dosen't start with a protocol add a default one;
+    if (input.match(/^[^:]:\/\//)) {
```

Comment:
im not sure if this regex is correct. Cause if i do `example.com".match(/^[^:]:\/\//)`, it returns null. But we want it to come in the if block in this case. Am i missing something?